### ISSUE TITLE
OntologyMap regex conditions

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Starcounter#2607]
I have the following mapping in my project:

```csharp
Polyjuice.OntologyMap("/people/partials/contacts/@w", "/so/person/@w", ... );
```

This does not let me use /people/partials/contacts/ url  to display page with list of customers.
There should be some condition to map only those urls which actually have @w value.

**Comment by warpech** [comment]
I agree that our URL matcher is too sensitive. In traditional way to match URLs, `/people/partials/contacts/` equals `/people/partials/contacts`. I guess this topic needs further research. Important factor for us is balancing performance vs developer-friendliness.

Some resources for the research:
- http://stackoverflow.com/questions/5948659/trailing-slash-in-urls-which-style-is-preferred
- https://cdivilly.wordpress.com/2014/03/11/why-trailing-slashes-on-uris-are-important/
- http://serverfault.com/questions/394757/should-my-web-server-add-trailing-slashes-to-urls-or-remove-them
- http://googlewebmastercentral.blogspot.se/2010/04/to-slash-or-not-to-slash.html
- http://www.searchenginejournal.com/linking-issues-why-a-trailing-slash-in-the-url-does-matter/13021/
